### PR TITLE
[Crypto] fix race conditions in DKG tests

### PR DESCRIPTION
**Pull request by tarakby**
Addresses https://github.com/dapperlabs/flow-internal/issues/1700
It fixes race conditions in DKG tests. All tests using `--race` flag are now passing.

This is a follow up on the work that started by https://github.com/onflow/flow-go/pull/2293

### CODE REVIEW

**Comment by tarakby** [comment]
> Were these tests reported as flaky by the Flaky Test Monitor?

No, they were not flakey. But running them with the race detector (`--race`) was showing race conditions. I guess the Flakey Test Monitor doesn't look specifically at race conditions for now right?

File: crypto/dkg_test.go [hunk 1]
```diff
@@ -1,3 +1,4 @@
+//go:build relic
 // +build relic
```
Comment (jordanschalm) [inline crypto/dkg_test.go:2]: Do we need both of these?

**Review by jordanschalm — APPROVED** [review]

**Review by kc1116 — APPROVED** [review]

File: crypto/dkg_test.go (hunk 1 above)
Comment (tarakby) [reply crypto/dkg_test.go:2]: IDE has been adding them automatically lately, though the `// +` seems enough.

**Comment by tarakby** [comment]
bors merge

**Pull request merged**